bin/prover-client/src/prover.rs

```diff
@@ -3,13 +3,15 @@ use std::{
     sync::{Arc, RwLock},
 };
 
-use strata_db::traits::{ProverDatabase, ProverTaskDatabase};
+use strata_db::traits::{ProofDatabase, ProverDatabase};
+use strata_primitives::proof::{ProofContext, ProofKey, ProofZkVm};
 use strata_proofimpl_evm_ee_stf::ELProofInput;
 use strata_rocksdb::{
     prover::db::{ProofDb, ProverDB},
```

Comment:
These two structs are likely redundant, could you please verify and merge them into one?